airflow/providers/amazon/aws/links/base.py

```diff
@@ -0,0 +1,87 @@
+#
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+from datetime import datetime
+from typing import TYPE_CHECKING, ClassVar, Optional
+
+from airflow.models import BaseOperatorLink, XCom
+
+if TYPE_CHECKING:
+    from airflow.models import BaseOperator
+    from airflow.models.taskinstance import TaskInstanceKey
+    from airflow.utils.context import Context
+
+
+BASE_AWS_CONSOLE_LINK = "https://console.{aws_domain}"
+
+
+class BaseAwsLink(BaseOperatorLink):
+    """Base Helper class for constructing AWS Console Link"""
+
+    name: ClassVar[str]
+    key: ClassVar[str]
+    format_str: ClassVar[str]
+
+    @staticmethod
+    def get_aws_domain(aws_partition) -> Optional[str]:
+        if aws_partition == "aws":
+            return "aws.amazon.com"
+        elif aws_partition == "aws-cn":
+            return "amazonaws.cn"
+        elif aws_partition == "aws-us-gov":
+            return "amazonaws-us-gov.com"
+
+        return None
+
+    def get_link(
+        self,
+        operator,
+        dttm: Optional[datetime] = None,
```

Comment:
I'm not crazy about `dttm` as a variable name, it's not immediately clear to me what ti is used for, but if changing that is a breaking change, then I guess we can live with it.

Reply:
Oh I grab it from other external links

In BaseOperatorLink mention that this signature is deprecated, need to check from which version of Airflow it deprecated and remove everything related to `dttm`
https://github.com/apache/airflow/blob/047a6162b0b4cbf07fe2fd978e335839a7d3900b/airflow/models/baseoperator.py#L1771-L1782

Reply:
> Oh I grab it from other external links

Yeah, I've seen `dttm` and similarly abbreviated variable names used pretty commonly in the Airflow codebase.  I guess it's just personal preference.

Reply:
I found this PR #21798 this is uses for support Airflow 2.2 and lower. So as soon as amazon-provider will required 2.3+ we could keep only `ti_key` and `operator`